designs/extensible-cli-and-scaffolding-plugins.md

```diff
@@ -0,0 +1,144 @@
+# Extensible CLI and Scaffolding Plugins
+
+## Overview 
+I would like for Kubebuilder to become more extensible, such that it could be imported and used as a library in other projects. Specifically, I'm looking for a way to use Kubebuilder's existing CLI and scaffolding for Go projects, but to also be able to augment Kubebuilder's project versions with other custom project versions so that I can support the Kubebuilder workflow with non-Go operators (e.g. operator-sdk's Ansible and Helm-based operators).
```

Comment:
One observation: I think the proposal is mixing `extensibility` with `using KB CLI as library`. I am wondering if decoupling them will simplify the proposal (if possible).

Reply:
To me these things go hand in hand, since `using KB CLI as a library` by itself is easy to do (CLI option structs like `cmd.apiOptions` could be exported) while making KB `extensible` in the manner suggested gives you a CLI library "for free". Does that make sense @droot?